File: src/waia/main/kotlin/banned/BanController.kt
package waia.main.kotlin.banned

import io.javalin.Context
import waia.main.kotlin.utils.Status

object BanController {

    private val banDao = BanDao()

    fun create(ctx: Context){
        val ban = ctx.body<Ban>()
        val status = banDao.save(ipAddress = ban.ipAddress)
        ctx.status(if (status.success) 200 else 400)
        ctx.json(status)
    }

    fun delete(ctx: Context){
        val resourceId = ctx.pathParam("ip")
        val status = banDao.delete(resourceId)
        ctx.json(status)
    }

    fun getAll(ctx: Context){
        ctx.json(banDao.findAll())
    }

    fun getOne(ctx: Context){
        val resourceId = ctx.pathParam("ip")
        val ban = banDao.findByIpAddress(ipAddress = resourceId)
        if (ban != null){
            ctx.json(ban)
        } else {
            ctx.json(Status(success = false, message = "Not found"))
        }
    }
}

File: app/controllers/BanController.scala
package controllers

import com.google.inject.Inject
import models.Bans
import play.api.libs.json.Json
import play.api.mvc.{AbstractController, Action, ControllerComponents}

class BanController @Inject()(cc: ControllerComponents) extends AbstractController(cc) {

  def findAll = Action(Ok(Json.toJson(Bans.findAll)))

  def findById(id: Long) = Action {
    val ban = Bans.findById(id)
    if (ban == null) {
      NotFound(Json.toJson(utils.Status(success = false, message = "Not found")))
    } else {
      Ok(Json.toJson(ban))
    }
  }

  def deleteById(ipAddress: String) = Action {
    val ban = Bans.delete(ipAddress)
    Ok(Json.toJson(ban))
  }

  def create: Action[Bans] = Action(parse.json[Bans]) { request =>
    val body = request.body
    val ban = Bans.create(body)
    Ok(Json.toJson(ban))
  }


}
